src/ecdsa/ellipticcurve.py

```diff
@@ -174,12 +175,23 @@ def __init__(self, curve, x, y, z, order=None, generator=False):
             self.__y = y
             self.__z = z
             self.__order = order
+        self.__generator = generator
         self.__precompute = []
-        if generator:
+
+        # when enabled, multiplication tables are precomputed only when actually needed:
+        # that is, the specific curve is actually first used in user code. otherwise, multiplication
+        # tables for _all_ curves supported are precomputed at library _import_ time effectively
+        ENABLE_LAZY_PRECOMPUTE = os.environ.get('PYTHON_ECDSA_ENABLE_LAZY_PRECOMPUTE', None) is not None
```

Comment:
I don't like the environment variable idea, this only makes the use of the library more complex, and delayed precomputation benefits everybody that doesn't use all curves every time

Reply:
note sure I understand: so should I remove the controlling flag altogether and make it "always lazy"?

Reply:
yes, make it always lazy